src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -322,13 +319,30 @@ public void bindServiceInstance(String applicationName, String serviceInstanceNa
     @Override
     public void bindServiceInstance(String applicationName, String serviceInstanceName, Map<String, Object> parameters) {
         UUID applicationGuid = getRequiredApplicationGuid(applicationName);
-        UUID serviceInstanceGuid = getRequiredServiceInstanceGuid(serviceInstanceName);
-        delegate.serviceBindingsV2()
-                .create(CreateServiceBindingRequest.builder()
-                                                   .applicationId(applicationGuid.toString())
-                                                   .serviceInstanceId(serviceInstanceGuid.toString())
-                                                   .parameters(parameters)
-                                                   .build())
+        var serviceResource = getServiceInstanceByName(serviceInstanceName).block();
+        if (serviceResource == null) {
+            throw new CloudOperationException(HttpStatus.NOT_FOUND, "Not Found", "Service instance " + serviceInstanceName + " not found.");
+        }
+        var request = CreateServiceBindingRequest.builder()
+                                                 .type(ServiceBindingType.APPLICATION)
+                                                 .parameters(parameters)
+                                                 .relationships(ServiceBindingRelationships.builder()
+                                                                                           .application(buildToOneRelationship(applicationGuid))
+                                                                                           .serviceInstance(buildToOneRelationship(UUID.fromString(serviceResource.getId())))
+                                                                                           .build())
+                                                 .build();
+
+        if (serviceResource.getType() == ServiceInstanceType.USER_PROVIDED) {
+            delegate.serviceBindingsV3()
+                    .create(request)
+                    .block();
+            return;
+        }
+        delegate.serviceBindingsV3()
+                .create(request)
+                .map(response -> response.getJobId()
+                                         .get())
```

Comment:
Can you use something like:
.create(request)
.filter(response -> response.getJobId().isPresent()) ..

Thus skipping the user_provided condition

Reply:
Good idea